chain/chain/src/chain.rs

```diff
@@ -2877,6 +2887,28 @@ impl<'a> ChainUpdate<'a> {
         )
     }
 
+    fn get_split_state_roots(
+        &mut self,
+        block: &Block,
+        shard_id: ShardId,
+    ) -> Result<HashMap<ShardUId, StateRoot>, Error> {
+        let next_shard_layout =
+            self.runtime_adapter.get_shard_layout(block.header().next_epoch_id())?;
+        let new_shards = next_shard_layout.get_split_shards(shard_id).expect(&format!(
+            "shard layout must contain maps of all shards to its split shards {}",
+            shard_id
+        ));
```

Comment:
nitpick: this will allocate an error string message even on the happy path. The more frugal way to write this would be

```rust
        let new_shards = next_shard_layout.get_split_shards(shard_id).unwrap_or_else( || panic!(
            "shard layout must contain maps of all shards to its split shards {}",
            shard_id
        ));
```